app/app_main.c

```diff
@@ -3761,7 +3714,12 @@ static ACVP_RESULT app_kas_ecc_handler(ACVP_TEST_CASE *test_case)
         FIPS_digest(Z, Zlen, (unsigned char *)tc->chash, NULL, md);
         tc->chashlen = M_EVP_MD_size(md);
     }
-    OPENSSL_cleanse(Z, Zlen);
+    rv = ACVP_SUCCESS;
+
+error:
+    if (Z) {
+        OPENSSL_cleanse(Z, Zlen);
+    }
     OPENSSL_free(Z);
     EC_KEY_free(ec);
```

Comment:
Do you need to check "if (ec) EC_KEY_free(ec);" ?

Reply:
no, only reason Z needed that check was since it was a cleanse and not a free